src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/services/DraftAdjudicationService.kt

```diff
@@ -57,27 +67,55 @@ class DraftAdjudicationService(val draftAdjudicationRepository: DraftAdjudicatio
     val draftAdjudication = draftAdjudicationRepository.findById(id).orElseThrow { throwEntityNotFoundException(id) }
 
     if (draftAdjudication.incidentDetails == null)
-      throw EntityNotFoundException("DraftAdjudication does not have any incident details to update")
+      throw EntityNotFoundException("DraftAdjudication does not include an incident statement")
 
     locationId?.let { draftAdjudication.incidentDetails?.locationId = it }
     dateTimeOfIncident?.let { draftAdjudication.incidentDetails?.dateTimeOfIncident = it }
 
-    return draftAdjudication.toDto()
+    return draftAdjudicationRepository
+      .save(draftAdjudication)
+      .toDto()
+  }
+
+  @Transactional
+  fun editIncidentStatement(id: Long, statement: String): DraftAdjudicationDto {
+    val draftAdjudication = draftAdjudicationRepository.findById(id).orElseThrow { throwEntityNotFoundException(id) }
+
+    if (draftAdjudication.incidentStatement == null)
+      throw EntityNotFoundException("DraftAdjudication does not have any incident statement to update")
+
+    if (authenticationFacade.currentUsername != draftAdjudication.incidentStatement?.createdByUserId)
+      throw UnAuthorisedToEditIncidentStatementException()
+
+    draftAdjudication.incidentStatement?.statement = statement
```

Comment:
Might be missing something, but i don't see any null checks for the statement

Reply:
So at this point **draftAdjudication.incidentStatement** is guaranteed to have a value. And the incoming statement parameter cannot be null as it's a val